python/python/bystro/ancestry/listener.py

```diff
@@ -38,30 +106,99 @@ def _load_queue_conf(queue_conf_path: str) -> QueueConf:
     return QueueConf(addresses=beanstalk_conf["addresses"], tubes=beanstalk_conf["tubes"])
 
 
+def _fill_missing_data(genotypes: pd.DataFrame) -> tuple[pd.DataFrame, pd.Series]:
```

Comment:
I'm a little confused why this imputation step is happening in the listener script - I think it should be part of preprocess or train?

Reply:
This is for the inference step, so the genotype matrix we're filling out here won't be available at train time.  Certainly open to suggestions about how to make that distinction clearer in the code.